### DISCUSSION TITLE
Hot to avoid caret and tilda in subdependencies of yarn.lock?

### DISCUSSION THREAD

**Discussion by hibace** [yarnpkg/berry#4260] [Q&A] [answered]
There is a problem to avoid malicious code in libraries. There is a case when we have a concrete version of lib in package.json. but the same lib in yarn.lock has subdependence with caret. How to avoid caret and tilda in subdependencies of yarn.lock? Thanks
![image](https://user-images.githubusercontent.com/9449536/159876942-bd33313f-2be5-4101-804d-459f1eb4b0d7.png)

**Comment by RDIL** [comment, marked as answer]
It depends on the authors of those libraries to choose the dependency versions.  You could modify them with packageExtensions but in the long term that’s not going to be extremely viable as you add more dependencies.

**Reply by hibace** [reply]
@RDIL  thx
